### PR TITLE
Use *ResolvedValue in template variables

### PR DESCRIPTION
**Pull request by mvilks**

### CODE REVIEW

File: template/template.go [hunk 1]
```diff
@@ -170,10 +170,35 @@ func (t *Template) Execute(w io.Writer, input map[string]any) error {
...
+		default:
+			executer.variables[k] = def
+			continue
+		case string:
+			function = "string"
+		case float64, int:
+			function = "number"
+		}
+
+		f, ok := t.registry[function]
```
Comment (janishorsts) [inline template/template.go:190]: Why look up a function every time we iterate through _input_ range? we can do it before _for_ loop

Comment (mvilks) [reply template/template.go:190]: Each input value can be a different type. Do you suggest iterating through the input slice to gather types and then again to create ResolvedValues?

**Review by janishorsts — APPROVED** [review]

File: template/template.go (hunk 1 above)
Comment (janishorsts) [reply template/template.go]: We can access registry functions directly without looking them up.

Comment (mvilks) [reply template/template.go]: Using registry function directly.

**Pull request merged**